src/reduced_order/rom_test_location.cpp

```diff
@@ -0,0 +1,100 @@
+#include "rom_test_location.h"
+
+namespace PHiLiP {
+namespace ProperOrthogonalDecomposition {
+
+template <int dim, int nstate>
+ROMTestLocation<dim, nstate>::ROMTestLocation(const RowVectorXd& parameter, std::shared_ptr<ROMSolution<dim, nstate>> rom_solution)
+        : parameter(parameter)
+        , rom_solution(rom_solution)
+        , mpi_communicator(MPI_COMM_WORLD)
+        , mpi_rank(dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD))
+        , pcout(std::cout, mpi_rank==0)
+{
+    pcout << "Creating ROM test location..." << std::endl;
+    compute_FOM_to_initial_ROM_error();
+    initial_rom_to_final_rom_error = 0;
+    compute_total_error();
+    pcout << "ROM test location created. Error estimate updated." << std::endl;
+}
+
+template <int dim, int nstate>
+void ROMTestLocation<dim, nstate>::compute_FOM_to_initial_ROM_error(){
+    pcout << "Computing adjoint-based error estimate between ROM and FOM..." << std::endl;
+    dealii::LinearAlgebra::distributed::Vector<double> gradient(rom_solution->right_hand_side);
+    dealii::LinearAlgebra::distributed::Vector<double> adjoint(rom_solution->right_hand_side);
+
+    gradient = rom_solution->gradient;
+
+    Parameters::LinearSolverParam linear_solver_param;
+    linear_solver_param.linear_solver_type = Parameters::LinearSolverParam::direct;
+    solve_linear(*rom_solution->system_matrix_transpose, gradient*=-1.0, adjoint, linear_solver_param);
+
+    //Compute dual weighted residual
+    fom_to_initial_rom_error = 0;
+    fom_to_initial_rom_error = -(adjoint * rom_solution->right_hand_side);
```

Comment:
Since adjoint and gradient are distributed vectors, summing errors from all processors might be required (or maybe not because direct and Amesos solvers do it internally? I am not sure).

Reply:
I have double checked and the complete error is output, not the error on a given processor, therefore it is ok.